Issue building ugs-pendant (same results as closed issue #1245)

<!--
If you are asking a question rather than filing a bug or a feature request, try asking on the google group: https://groups.google.com/forum/#!forum/universal-gcode-sender

If you are experiencing problems, have a look at the troubleshooting guides: http://winder.github.io/ugs_website/guide/troubleshooting/
-->

## Feature request
<!-- If you have a new idea for a feature, describe it here. You can delete the rest of the form.  -->

## Problem description 
<!-- If you suspect that you've found a bug give it a short description here and fill in the rest of the form to make it easier for us to reproduce and identify the problem. -->
When attempting to run on Raspberry Pi, failing when building ugs-pendant.

### Expected Behavior
<!-- Example: When I try to open a Gcode file it should be shown in the visualizer -->
Expect a completed build

### Actual Behavior
<!-- Example: Another file is loaded and shown in the visualizer -->
[INFO] ------------------------------------------------------------------------
[INFO] Building ugs-pendant 2.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- frontend-maven-plugin:1.6:install-node-and-npm (install node and npm) @ ugs-pendant ---
[INFO] Installing node version v10.15.0
[INFO] Downloading https://nodejs.org/dist/v10.15.0/node-v10.15.0-linux-x86.tar.gz to /root/.m2/repository/com/github/eirslett/node/10.15.0/node-10.15.0-linux-x86.tar.gz
[INFO] No proxies configured
[INFO] No proxy was configured, downloading directly
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] ugs-parent ......................................... SUCCESS [  1.223 s]
[INFO] ugs-core ........................................... SUCCESS [ 10.816 s]
[INFO] ugs-pendant ........................................ FAILURE [  3.446 s]
[INFO] ugs-cli ............................................ SKIPPED
[INFO] ugs-classic ........................................ SKIPPED
[INFO] ugs-platform-parent ................................ SKIPPED
[INFO] ugs-platform-branding .............................. SKIPPED
[INFO] ugs-platform-ugslib ................................ SKIPPED
[INFO] ugs-platform-ugscore ............................... SKIPPED
[INFO] ugs-platform-visualizer ............................ SKIPPED
[INFO] ugs-platform-plugin-workflow ....................... SKIPPED
[INFO] ugs-platform-gcode-editor .......................... SKIPPED
[INFO] ugs-platform-surfacescanner ........................ SKIPPED
[INFO] ProbeModule ........................................ SKIPPED
[INFO] DowelModule ........................................ SKIPPED
[INFO] GcodeTools ......................................... SKIPPED
[INFO] ugs-platform-plugin-jog ............................ SKIPPED
[INFO] ugs-platform-plugin-setup-wizard ................... SKIPPED
[INFO] ugs-platform-welcome-page .......................... SKIPPED
[INFO] ugs-platform-app ................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 19.447 s
[INFO] Finished at: 2019-08-22T16:20:34+00:00
[INFO] Final Memory: 22M/52M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.6:install-node-and-npm (install node and npm) on project ugs-pendant: Could not download Node.js: Got error code 404 from the server. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :ugs-pendant

### Steps to Reproduce the Problem
<!-- Example:
  1. Open the program
  1. Click on the toolbar menu icon for opening a file
  1. Selects a file in the dialog and click ok
  1. The visualizer shows the wrong file and the filename in the window title is wrong
-->
  1. Virgin install of Pi.  Added Java, maven, etc.  Fails every time.
 

### Specifications
#### Version
<!-- 
  What versions of the application does the issue exist in.  You find the version 
  in the title bar of the application. Example:
  
  UGS Platform 2.0 - Nightly Build Mar 23, 2018
  UGS Classic 1.0.9
-->

Latest build.

#### Operating system
<!-- 
  What operating system are you running. Example: 
  Windows 10
  Linux Ubuntu
  Linux on RaspberryPI
  Mac OSX
-->

Linux on RaspberryPI
#### Platform
<!-- 
  What hardware are you using. Example:
  GRBL 1.1f
  TinyG
  SmoothieBoard 
-->

Pi 3b

#### Other
<!-- Please provide screen shots, gcode-files, output from $$-command that you think may help us to  identify the problem. -->

mvn -version
Apache Maven 3.3.9
Maven home: /usr/share/maven
Java version: 1.8.0_222, vendor: Raspbian
Java home: /usr/lib/jvm/java-8-openjdk-armhf/jre
Default locale: en_GB, platform encoding: UTF-8
OS name: "linux", version: "4.9.80-v7+", arch: "arm", family: "unix"

I have the same issue.
I am trying to install Universal-G-Code-Sender on a Pi 2.
It is running Linux raspberrypi 4.9.80-V7
My Node.js version is:
My Mavin version is: 3.6.2
Java: 1.8.0_65

This is the first time I've installed the software so I don't know what to expect.

When I tried to use the command you mentioned: mvn install -P'!create-pendant-ui-web'

I eventually saw:
[INFO] Installing node version v10.15.0
[INFO] Downloading https://nodejs.org/dist/v10.15.0/node-v10.15.0-linux-x86.tar.gz to /home/pi/.m2/repository/com/github/eirslett/node/10.15.0/node-10.15.0-linux-x86.tar.gz
[INFO] No proxies configured
[INFO] No proxy was configured, downloading directly
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for ugs-parent 2.0-SNAPSHOT:
[INFO] 
[INFO] ugs-parent ......................................... SUCCESS [  2.971 s]
[INFO] ugs-core ........................................... SUCCESS [02:51 min]
[INFO] ugs-pendant ........................................ FAILURE [  7.378 s]
[INFO] ugs-cli ............................................ SKIPPED
[INFO] ugs-classic ........................................ SKIPPED
[INFO] ugs-platform-parent ................................ SKIPPED
[INFO] ugs-platform-branding .............................. SKIPPED
[INFO] ugs-platform-ugslib ................................ SKIPPED
[INFO] ugs-platform-ugscore ............................... SKIPPED
[INFO] ugs-platform-visualizer ............................ SKIPPED
[INFO] ugs-platform-plugin-workflow ....................... SKIPPED
[INFO] ugs-platform-gcode-editor .......................... SKIPPED
[INFO] ugs-platform-surfacescanner ........................ SKIPPED
[INFO] ProbeModule ........................................ SKIPPED
[INFO] DowelModule ........................................ SKIPPED
[INFO] GcodeTools ......................................... SKIPPED
[INFO] ugs-platform-plugin-jog ............................ SKIPPED
[INFO] ugs-platform-plugin-setup-wizard ................... SKIPPED
[INFO] ugs-platform-welcome-page .......................... SKIPPED
[INFO] ugs-platform-app ................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:11 min
[INFO] Finished at: 2019-10-10T13:14:17-04:00
[INFO] ------------------------------------------------------------------------
[WARNING] The requested profile "| create-pendant ui-web" could not be activated because it does not exist.
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.6:install-node-and-npm (install node and npm) on project ugs-pendant: Could not download Node.js: Got error code 404 from the server. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :ugs-pendant

Any ideas on what to do?

--- a/new-module-archetype/src/main/resources/archetype-resources/pom.xml
+++ b/new-module-archetype/src/main/resources/archetype-resources/pom.xml
@@ -88,7 +88,7 @@
     <dependency> 
       <groupId>org.mockito</groupId>  
       <artifactId>mockito-core</artifactId>  
-      <version>2.7.22</version>  
+      <version>${mockito.version}</version>
       <scope>test</scope>  
       <type>jar</type> 
     </dependency>  

--- a/pom.xml
+++ b/pom.xml
@@ -51,9 +51,11 @@
     <vecmath.version>1.5.2</vecmath.version>
     <zxing.version>3.2.0</zxing.version>
     <qrgen.version>1.3</qrgen.version>
+    <jacoco.version>0.8.5</jacoco.version>
     <swing-layout.version>1.0.3</swing-layout.version>
     <surefire.reportplugin.version>2.19.1</surefire.reportplugin.version>
     <download-maven-plugin.version>1.3.0</download-maven-plugin.version>
+    <mockito.version>3.1.0</mockito.version>
 
     <ugs.maven-compiler-plugin.version>3.6.1</ugs.maven-compiler-plugin.version>
     <ugs.jvm.version>1.8</ugs.jvm.version>
@@ -181,7 +183,7 @@
     <dependency>
       <groupId>org.mockito</groupId>
       <artifactId>mockito-core</artifactId>
-      <version>2.7.22</version>
+      <version>${mockito.version}</version>
       <scope>test</scope>
       <type>jar</type>
     </dependency>

--- a/ugs-core/pom.xml
+++ b/ugs-core/pom.xml
@@ -159,7 +159,7 @@
       <plugin>
         <groupId>org.jacoco</groupId>
         <artifactId>jacoco-maven-plugin</artifactId>
-        <version>0.7.5.201505241946</version>
+        <version>${jacoco.version}</version>
         <executions>
           <execution>
             <goals>

--- a/ugs-platform/DowelModule/pom.xml
+++ b/ugs-platform/DowelModule/pom.xml
@@ -92,7 +92,7 @@
     <dependency> 
       <groupId>org.mockito</groupId>  
       <artifactId>mockito-core</artifactId>  
-      <version>2.7.22</version>  
+      <version>${mockito.version}</version>
       <scope>test</scope>  
       <type>jar</type> 
     </dependency>  

--- a/ugs-platform/GcodeTools/pom.xml
+++ b/ugs-platform/GcodeTools/pom.xml
@@ -88,7 +88,7 @@
     <dependency> 
       <groupId>org.mockito</groupId>  
       <artifactId>mockito-core</artifactId>  
-      <version>2.7.22</version>  
+      <version>${mockito.version}</version>
       <scope>test</scope>  
       <type>jar</type> 
     </dependency>  

--- a/ugs-platform/ProbeModule/pom.xml
+++ b/ugs-platform/ProbeModule/pom.xml
@@ -91,7 +91,7 @@
       <dependency>
         <groupId>org.mockito</groupId>
         <artifactId>mockito-core</artifactId>
-        <version>2.7.22</version>
+        <version>${mockito.version}</version>
         <scope>test</scope>
         <type>jar</type>
       </dependency>

--- a/ugs-platform/ugs-platform-plugin-setup-wizard/pom.xml
+++ b/ugs-platform/ugs-platform-plugin-setup-wizard/pom.xml
@@ -99,7 +99,7 @@
         <dependency>
             <groupId>org.mockito</groupId>
             <artifactId>mockito-core</artifactId>
-            <version>2.7.22</version>
+            <version>${mockito.version}</version>
             <scope>test</scope>
             <type>jar</type>
         </dependency>

--- a/ugs-platform/ugs-platform-welcome-page/pom.xml
+++ b/ugs-platform/ugs-platform-welcome-page/pom.xml
@@ -114,7 +114,7 @@
     <dependency> 
       <groupId>org.mockito</groupId>  
       <artifactId>mockito-core</artifactId>  
-      <version>2.7.22</version>  
+      <version>${mockito.version}</version>
       <scope>test</scope>  
       <type>jar</type> 
     </dependency>  

